server/src/main/java/com/exacaster/lighter/rest/SessionController.java

```diff
@@ -91,9 +93,9 @@ public Optional<Object> getLog(@PathVariable String id, @Nullable @Header("X-Com
     }
 
     @Post("/{id}/statements")
-    @Status(HttpStatus.CREATED)
-    public Statement postStatements(@PathVariable String id, @Valid @Body Statement statement) {
-        return sessionService.createStatement(id, statement);
+    public HttpResponse postStatements(@PathVariable String id, @Valid @Body Statement statement) {
+        MicronautStatementCreationResultToResponseMapper resultToResponseMapper = new MicronautStatementCreationResultToResponseMapper();
```

Comment:
It shouldn't be needed to create a new instance of the mapper for each `postStatement` request. Maybe it could be annotated with `@Singleton` and injected through controllers constructor instead?

Reply:
can do. i would not consider it as a big memory/performance footprint but can make it `@Singleton`